docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -310,31 +321,59 @@ def a_job():
 
 If you have specific requirements for where and how your outputs should be stored and retrieved, you can define your own IO manager. This boils down to implementing two functions: one that stores outputs and one that loads inputs.
 
-To define an IO manager, use the <PyObject module="dagster" object="io_manager" displayText="@io_manager" /> decorator.
+To define an IO manager, you will want to extend the <PyObject module="dagster" object="IOManager" /> class. Often times, you will want to extend the <PyObject module="dagster" object="ConfigurableIOManager"/> class (which subclasses `IOManager`) to attach a config schema to your IO manager.
+
+Here, we define a simple IO manager that reads and writes CSV values to the filesystem. It takes an optional prefix path through config.
 
 ```python file=/concepts/io_management/custom_io_manager.py startafter=start_io_manager_marker endbefore=end_io_manager_marker
-from dagster import IOManager, io_manager
+from dagster import ConfigurableIOManager
 
 
-class MyIOManager(IOManager):
+class MyIOManager(ConfigurableIOManager):
+    path_prefix: List[str] = []
+
     def _get_path(self, context) -> str:
-        return "/".join(context.asset_key.path)
+        return "/".join(self.path_prefix + context.asset_key.path)
 
     def handle_output(self, context, obj):
         write_csv(self._get_path(context), obj)
 
     def load_input(self, context):
         return read_csv(self._get_path(context))
+```
 
+The provided `context` argument for `handle_output` is an <PyObject module="dagster" object="OutputContext" />. The provided `context` argument for `load_input` is an <PyObject module="dagster" object="InputContext" />. The linked API documentation lists all the fields that are available on these objects.
 
-@io_manager
-def my_io_manager():
-    return MyIOManager()
-```
+### Using an IO manager factory
 
-The <PyObject module="dagster" object="io_manager" /> decorator behaves nearly identically to the <PyObject module="dagster" object="resource" /> decorator. It yields an <PyObject module="dagster" object="IOManagerDefinition" />, which is a subclass of <PyObject module="dagster" object="ResourceDefinition" /> that will produce an <PyObject module="dagster" object="IOManager" />.
+If your IO manager is more complex, or needs to manage internal state, it may make sense to split out the IO manager definition from its configuration. In this case, you can use `ConfigurableIOManagerFactory`, which specifies config schema and implements a factory function that takes the config and returns an IO manager.
```

Comment:
> you can use `ConfigurableIOManagerFactory`

do we have apidoc for `ConfigurableIOManagerFactory`? maybe use PyObject instead of plain code syntax here so users can click to see details

Reply:
+1 ^

We should use this whenever possible. Better experience for users, but it'll also make it easy for us to keep things updated/correct.